src/transports/http.rs

```diff
@@ -119,15 +126,17 @@ impl Http {
                 }),
         );
 
+
         let basic_auth = {
             let url = Url::parse(url)?;
             let user = url.username();
 
             if user.len() > 0 {
-                Some(hyper::header::Basic {
-                    username: user.into(),
-                    password: url.password().map(Into::into),
-                })
+                let auth = match url.password() {
+                    Some(pass) => format!("{}:{}", user, pass),
+                    None => format!("{}:", user)
```

Comment:
you sure we need the trailing `:`? Maybe let's encode only the username.

Reply:
I was conflicted on this myself, since the HTTP Basic Auth RFC does not specify exactly what to do with an empty password, just that the format must be 'Basic username:password', with user:pass part encoded in Base64. I am not well-versed on HTTP spec, though, so I may be wrong on this, but figured it would be safest to do 'user:'

https://tools.ietf.org/html/rfc7617#section-2

My conclusion was no password just counts as an 'empty' password so it would be encoded as 'Basic username:'

Reply:
Ok, makes sense!